From a26768cc5bb452dd93c9227dfd432f4626e8369e Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Fri, 5 Apr 2013 00:18:34 +0200
Subject: [PATCH 09/65] r92su: don't start aggregation for critical data frames

This change should shorten the time it takes to connect
to a network.

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 r92su/tx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/wireless/r92su/tx.c
+++ b/drivers/net/wireless/r92su/tx.c
@@ -384,7 +384,7 @@ r92su_tx_add_80211(struct r92su *r92su,
 		tid = skb->priority % ARRAY_SIZE(ieee802_1d_to_ac);
 		qos_ctl[0] = tid;
 
-		if (tx_info->ht_possible &&
+		if (tx_info->ht_possible && !tx_info->low_rate &&
 		    skb_get_queue_mapping(skb) != IEEE80211_AC_VO) {
 			if (!bss_priv->tx_tid[tid].addba_issued) {
 				r92su_h2c_start_ba(r92su, tid);
